tracer/src/Datadog.Trace.SourceGenerators/Helpers/YamlReader.cs

```diff
@@ -0,0 +1,142 @@
+// <copyright file="YamlReader.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Text;
+
+namespace Datadog.Trace.SourceGenerators.Helpers
+{
+    /// <summary>
+    /// Simple YAML parser for reading documentation strings from YAML files.
+    /// Supports basic YAML features needed for configuration documentation.
+    /// </summary>
+    internal static class YamlReader
+    {
+        /// <summary>
+        /// Parses a YAML file containing configuration key documentation.
+        /// Expects format: KEY: | followed by multi-line documentation
+        /// </summary>
+        public static Dictionary<string, string> ParseDocumentation(string yamlContent)
+        {
+            var result = new Dictionary<string, string>();
+            var lines = yamlContent.Replace("\r\n", "\n").Split('\n');
```

Comment:
tiny nit, if this code ends up running a lot, that's a bunch of extra allocations, maybe we could make it a span first and use that? If it's only going to run once on project load, then it's less of an issue (just makes initial loads slower)